src/cow-react/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx

```diff
@@ -66,9 +67,9 @@ export function CurrencyInputPanel(props: CurrencyInputPanelProps) {
 
   const isSupportedNetwork = isSupportedChainId(props.chainId as number | undefined)
   const { priceImpact, loading: priceImpactLoading } = priceImpactParams || {}
-  const { field, currency, balance, fiatAmount, viewAmount, receiveAmountInfo } = currencyInfo
+  const { field, currency, balance, fiatAmount, rawAmount, isIndependent, receiveAmountInfo } = currencyInfo
   const disabled = props.disabled || !isSupportedNetwork
-
+  const viewAmount = formatInputAmount(rawAmount, balance, isIndependent)
```

Comment:
I get now your other TODO. 

Yes, I also remember to have mentioned this at some point. I agree on moving the responsibility

Reply:
maybe `rawAmountIsCalculated` or `rawAmountIsUserInputed`?

Reply:
`isDerivedField`?
`isCalculatedField`?

Reply:
Actually it's just amount, renamed it